File: Application/lib/domain/comment.dart
import 'dart:convert';

class Comment {
  String uuid;
  String author;
  String body;
  String targetUuid;
  String timestamp;

  Comment(
    this.uuid,
    this.author,
    this.body,
    this.targetUuid,
    this.timestamp,
  );

  factory Comment.fromJson(dynamic json) {
    return Comment(
      json['uuid'] as String,
      // json['children'] as List<Comment>,
      json['author'] as String,
      json['body'] as String,
      json['targetUuid'] as String,
      json['timestamp'] as String,
    );
  }

  @override
  String toString() {
    return jsonEncode({
      "authorUuid": author,
      "targetUuid": targetUuid,
      "body": body,
    });
  }
}

File: backend/src/main/java/com/bachelor/backend/domain/feed/Comment.java
package com.bachelor.backend.domain.feed;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.immutables.value.Value;

import java.io.Serializable;
import java.time.Instant;
import java.util.UUID;

@Value.Immutable
@JsonSerialize(as = ImmutableComment.class)
@JsonDeserialize(as = ImmutableComment.class)
public abstract class Comment implements Serializable {

    public static Comment initialize(UUID authorUuid, UUID targetUuid, String body) {
        return create(UUID.randomUUID(), authorUuid, targetUuid, body);
    }

    public static Comment create(UUID uuid, UUID authorUuid, UUID targetUuid, String body) {
        return ImmutableComment.builder()
                .uuid(uuid)
                .authorUuid(authorUuid)
                .targetUuid(targetUuid)
                .body(body)
                .timestamp(Instant.now().toString())
                .build();
    }

    public static Comment create(NewComment newComment) {
        return initialize(newComment.getAuthorUuid(), newComment.getTargetUuid(), newComment.getBody());
    }

    public ImmutableComment copy() {
        return ImmutableComment.copyOf(this);
    }

    public abstract UUID getUuid();

    public abstract UUID getAuthorUuid();

    public abstract UUID getTargetUuid();

    public abstract String getBody();

    public abstract String getTimestamp();

    @Value.Default
    public String getAuthor() {
        return "NaN";
    }

    @JsonIgnore
    public Instant getTime() {
        return Instant.parse(getTimestamp());
    }
}
